### DISCUSSION TITLE
Help debug my Auto sessions

### DISCUSSION THREAD

**Discussion by alankritjoshi** [rmagatti/auto-session#276] [Q&A] [answered]
My lazy plugin config is [here](https://github.com/alankritjoshi/dotfiles/blob/ea07b8ddb2a66accd58fa35bf81b4b9c1042b0fb/private_dot_config/nvim/lua/plugins/session.lua).

Expected:
If I use `<Leader>qq` to quit, I was expecting the last buffers to be saved and a session created.

Actual:
No session is created. I have to manually do SessionSave. 

 Am I missing something?

**Comment by cameronr** [comment]
Just saw this, that config should autosave (assuming you're not in one of the suppressed dirs). Can you set `log_level = 'debug'` and then run:

`:lua require('auto-session').AutoSaveSession()`

and then post what `:messages` and/or ':Notifications' shows?

**Reply by cameronr** [reply, marked as answer]
I just noticed that your config won't actually load `auto-session` because you've defined keys but haven't set `lazy = false` (or defined an event to trigger lazy loading). That means `Lazy.nvim` won't load `auto-session` until you press one of the keys. That's why it wasn't working for you. If you add `lazy = false` to your `auto-session` config, it should work.

**Reply by alankritjoshi** [reply]
Thanks!
